src/client/opamSwitchCommand.ml

```diff
@@ -312,6 +312,11 @@ let create
   let update_config = update_config && not (OpamSwitch.is_external switch) in
   let comp_dir = OpamPath.Switch.root gt.root switch in
   let simulate = OpamStateConfig.(!r.dryrun) || OpamClientConfig.(!r.show) in
+  (* TODO: We might want to make some more tests (quotes?, ...) *)
+  let switch_str = OpamSwitch.to_string switch in
+  if String.equal switch_str "-" then
+    OpamConsole.error_and_exit `Bad_arguments
+      "The name of the switch is not valid";
```

Comment:
This check need to be added in `OpamSwitch.check` function, that is called at each `OpamSwitch.of_string` (that is called by cli argument parsing, among other places.)